### PR TITLE
Add cluster registration deletion script for Fleet

### PR DESCRIPTION
**Pull request by weyfonk**
This is a cleanup script to work around a known Fleet bug whereby patching a downstream cluster, for instance when re-deploying a Fleet agent in such a cluster, causes new resources to be created without obsolete resources being deleted. Ultimately, this clutters the upstream cluster.

This script retrieves all cluster registration resources, orders them by cluster then by creation timestamp, and deletes all but the youngest cluster registration for each cluster. This causes obsolete cluster registrations and their child resources to be deleted.

### CODE REVIEW

File: fleet-delete-cluster-registration/delete_old_resources.sh
Comment (gionn) [inline fleet-delete-cluster-registration/delete_old_resources.sh:61]: this approach hammered my machine for minutes with thousands of kubectl process, I made it thru simply skipping parallelism:

```suggestion
    delete_chunk $i $last_chunk_idx
done
```

Comment (weyfonk) [reply fleet-delete-cluster-registration/delete_old_resources.sh:61]: You could most likely achieve something similar by increasing the chunk size (2nd script argument) to a very high value, up to or even over (eg. in case only an order of magnitude is known) the number of cluster registration resources you need to delete.

Comment (gionn) [reply fleet-delete-cluster-registration/delete_old_resources.sh:61]: didn't notice it, maybe better have a more conservatively low default, better discover how it's easy to make it faster than slower :D

**Comment by dnoland1** [comment]
I sometimes get `unary operator expected` errors when running the script:
```
./clusterregistration-cleanup.sh
Cluster: c-5zfq8Cluster: c-7f75dCluster: c-dbplkCluster: c-dbplkCluster: c-gdgsbCluster: c-7f75dCluster: c-fvbgkCluster: c-ndb9gCluster: c-pzst2Cluster: c-kjwnzCluster: c-ndb9gCluster: c-kjwnzCluster: c-5zfq8./clusterregistration-cleanup.sh: line 30: [: =: unary operator expected
	clusterregistration.fleet.cattle.io "request-fk24b" deleted
Cluster: c-fvbgk	clusterregistration.fleet.cattle.io "request-t6j99" deleted
```

**Comment by dnoland1** [comment]
Would also be nice to have a summary at the end that shows how many clusterregistration CRs were removed and how many are currently in the system

**Comment by manno** [comment]
> I sometimes get unary operator expected errors when running the script:

I opened https://github.com/weyfonk/support-tools/pull/1/files to so the loop doesn't exceed bounds when it reaches ahead.
This should fix the issue with the unary error message.

I think it's still safe to use the script, since we don't want to delete the last cluster and comparing the cluster name againt an empty string fails, so the delete is not done.

**Review by dnoland1 — APPROVED** [review]
I ran this script on various production environments, so I'm okay with it being merged into the repo.

**Comment by dkeightley** [comment]
@weyfonk I added a basic readme with your initial description, would you like to review/update this before merging?

**Comment by weyfonk** [comment]
> @weyfonk I added a basic readme with your initial description, would you like to review/update this before merging?

Thanks! I can't approve a PR which I authored, but this LGTM :+1: 
I would also be fine with the README being embedded at the beginning of the script.

**Pull request merged**
